src/CoreWCF.MSMQ/src/CoreWCF/Channels/MsmqTransportSecurity.cs

```diff
@@ -0,0 +1,97 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.ComponentModel;
+using System.Net.Security;
+
+namespace CoreWCF.Channels
+{
+    public sealed class MsmqTransportSecurity
+    {
+        private MsmqAuthenticationMode _msmqAuthenticationMode;
+        private MsmqEncryptionAlgorithm _msmqEncryptionAlgorithm;
+        private MsmqSecureHashAlgorithm _msmqHashAlgorithm;
+        private ProtectionLevel _msmqProtectionLevel;
+
+        public MsmqTransportSecurity()
+        {
+            _msmqAuthenticationMode = MsmqDefaults.MsmqAuthenticationMode1;
+            _msmqEncryptionAlgorithm = MsmqDefaults.MsmqEncryptionAlgorithm1;
+            _msmqHashAlgorithm = MsmqDefaults.MsmqSecureHashAlgorithm;
+            _msmqProtectionLevel = MsmqDefaults.MsmqProtectionLevel;
+        }
+
+        public MsmqTransportSecurity(MsmqTransportSecurity other)
+        {
+            if (null == other)
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("other");
+            _msmqAuthenticationMode = other.MsmqAuthenticationMode;
+            _msmqEncryptionAlgorithm = other.MsmqEncryptionAlgorithm;
+            _msmqHashAlgorithm = other.MsmqSecureHashAlgorithm;
+            _msmqProtectionLevel = other.MsmqProtectionLevel;
+        }
+
+        internal bool Enabled
+        {
+            get
+            {
+                return _msmqAuthenticationMode != MsmqAuthenticationMode.None && _msmqProtectionLevel != ProtectionLevel.None;
+            }
+        }
+
+        [DefaultValue(MsmqDefaults.MsmqAuthenticationMode1)]
+        public MsmqAuthenticationMode MsmqAuthenticationMode
+        {
+            get { return _msmqAuthenticationMode; }
+            set
+            {
+                //if (!MsmqAuthenticationModeHelper.IsDefined(value))
+                //    throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new ArgumentOutOfRangeException("value"));
```

Comment:
The value does need to be validated as you can cast any integer to an enum type.